### PR TITLE
Make ScubaResults.json the default JSON output

### LINKED ISSUE #1023: Make Invoke-ScubaCached backwards compatible with ScubaResults.json
# 💡 Summary #

RunCached is a messy function that violates several different  programming principles but was created to do some neat little ScubaGear tricks. 
Following in the same spirit, this issue is to add functionality to ScubaCached to be able to convert `ProviderSettingsExport.json` to `ScubaResults.json` format.  See #970 for specifics on the ScubaResults.json

## Motivation and context ##

For our own internal use.

## Implementation notes ##
Modification of ScubaCached with the `-MergeJson` parameter.

## Acceptance criteria ##

When we are able to convert `ProviderSettingsExport.json` to `ScubaResults.json` on demand.

### LINKED ISSUE #1205: Make MergeJson the default behavior
# 💡 Summary #
Make it so that the various json files output by ScubaGear are merged together by default (see https://github.com/cisagov/ScubaGear/blob/main/docs/configuration/parameters.md#mergejson).

## Motivation and context ##
Having a single file that encompasses all ScubaGear output would increase the portability of ScubaGear reports.

## Implementation notes ##
Dependent on https://github.com/cisagov/ScubaGear/issues/1023.

### PR DESCRIPTION
**Pull request by buidav**
## 🗣 Description ##

- Makes ScubaResults.json the default output for ScubaGear
- Inverts the `-MergeJSON` switch to `-KeepIndividualJSON` parameter. 
- Make `ScubaCached` backwards compatible with `ScubaResults.json`
- Update any tests that need it to match the new ScubaGear output behavior. 

TODO: 
- Update the functional and smoke tests to work with the ScubaResults.json in #1317 

## 💭 Motivation and context ##

Closes #1205
Closes #1023 

## 🧪 Testing ##

- Run `Invoke-SCuBA`. Observe `ScubaResults.json` is the default output.
- Run `Invoke-SCuBA -KeepIndividualJSON`. Observe `ProviderSettingsExport.json` is output instead of ScubaResutls.
- Run `Invoke-SCuBACached`. Observe `ScubaResults.json` is the default output.
- Run `Invoke-SCuBACached -ExportProvider $false`. Observe `ScubaCached` still performs as expected.
- Run `Invoke-SCuBACached` with the `-KeepIndividualJSON` flag. Observe performance benefit with this flag. Along with no change in behavior. 
- Run functional tests for a product as see no breaking changes.

- For reviewers and merge coordinator **Check smoke tests or other workflows to see if there are errors from the new output behavior**.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: docs/configuration/parameters.md
Comment (adhilto) [inline docs/configuration/parameters.md:1]: A few surviving instances of the `MergeJson` flag in this file:
image
image

File: PowerShell/ScubaGear/Modules/Orchestrator.psm1
Comment (adhilto) [inline PowerShell/ScubaGear/Modules/Orchestrator.psm1:1]: A surviving instance of `MergeJson`
image

**Review by adhilto — COMMENTED** [review]
Some minor things:

File: PowerShell/ScubaGear/Modules/Orchestrator.psm1 [hunk 1]
```diff
@@ -1733,7 +1733,19 @@ function Invoke-SCuBACached {
                 }
                 Invoke-ProviderList @ProviderParams
             }
+
             $FileName = Join-Path -Path $OutPath -ChildPath "$($OutProviderFileName).json"
+            if (-not (Test-Path $FileName)) {
+                # When running Invoke-ScubaCached, the provider output might not exist as a stand-alone
+                # file depending on what version of ScubaGear created the output. If the provider output
+                # does not exist as a stand-alone file, create it from the ScubaRestuls file so the other functions
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/Modules/Orchestrator.psm1:1741]:
```suggestion
                # does not exist as a stand-alone file, create it from the ScubaResults file so the other functions
```

File: docs/execution/reports.md [hunk 2]
```diff
@@ -6,8 +6,8 @@ When ScubaGear runs, it creates a new time-stamped subdirectory wherein it will
...
-| `ScubaResults.json`           | This JSON file encapsulates all ScubaGear output in a format can could be automatically parsed by a downstream system. It contains metadata about the run and the tenant, summary counts of the test results, the test results, and the raw provider output. Only present if ScubaGear is run with the `MergeJson` flag. |
+| `ProvideSettingsExport.json` | This JSON file contains all of the information that ScubaGear extracted from the products.  A highly-motivated admin might find this useful for understanding how ScubaGear arrived at its results. Only present if ScubaGear is run with the `KeepIndividualJson` flag; if run without the `KeepIndividualJSON` parameter, the contents of this file will be merged into the ScubaResults.json file. |
+| `ScubaResults.json`           | This JSON file encapsulates all ScubaGear output in a format can could be automatically parsed by a downstream system. It contains metadata about the run and the tenant, summary counts of the test results, the test results, and the raw provider output. Not present if ScubaGear is run with the `KeepIndividualJSON` flag. |
```
Comment (mitchelbaker-cisa) [inline docs/execution/reports.md:10]:
```suggestion
| `ScubaResults.json`           | This JSON file encapsulates all ScubaGear output in a format that is automatically parsed by a downstream system. It contains metadata about the run and the tenant, summary counts of the test results, the test results, and the raw provider output. Not present if ScubaGear is run with the `KeepIndividualJSON` flag. |
```

File: Testing/Functional/SmokeTest/SmokeTest001.Tests.ps1 [hunk 3]
```diff
@@ -51,11 +51,11 @@ Describe "Smoke Test: Generate Output" {
     Context "Invoke Scuba for $Organization" {
         BeforeAll {
             if ($PSCmdlet.ParameterSetName -eq 'Manual'){
-                { Invoke-SCuBA -ProductNames "*" -M365Environment $M365Environment -Quiet} |
+                { Invoke-SCuBA -ProductNames "*" -M365Environment $M365Environment -Quiet -KeepIndividualJSON} |
```
Comment (mitchelbaker-cisa) [inline Testing/Functional/SmokeTest/SmokeTest001.Tests.ps1:54]: Close #1193 if we plan to keep individual JSON

**Review by mitchelbaker-cisa — APPROVED** [review]
Ran all of these successfully with no issue. A few small comments to add is all. 

- [x] Run Invoke-SCuBA. Observe ScubaResults.json is the default output.
- [x] Run Invoke-SCuBA -KeepIndividualJSON. Observe ProviderSettingsExport.json is output instead of ScubaResutls.
- [x] Run Invoke-SCuBACached. Observe ScubaResults.json is the default output.
- [x] Run Invoke-SCuBACached -ExportProvider $false. Observe ScubaCached still performs as expected.
- [x] Run Invoke-SCuBACached with the -KeepIndividualJSON flag. Observe performance benefit with this flag. Along with no change in behavior.
- [x] Run functional tests for a product as see no breaking changes (ran aad functional tests successfully)

**Review by adhilto — APPROVED** [review]
Looks great. A few final notes:

- I manually ran the nightly functional tests, they are all passing on this branch.
- The one failure from the smoke test is related to the encoding issue, fixed in https://github.com/cisagov/ScubaGear/pull/1302, so not applicable here.

**Pull request merged**